### DISCUSSION TITLE
Add search with whole word options, future find-all window

### DISCUSSION THREAD

**Discussion by catherdersoftware** [sumatrapdfreader/sumatrapdf#4200] [Ideas]
I love SumatraPDF for its lightweight footprint as I usually have a lot of PDF documents open.  The one feature I miss the most is a "whole word" search function, i.e., searching "model" as a whole world will not match "models" and "modeling". 

I would also like to see a new window for "find-all" results so I can bounce around a document. I understand that this would be a lot of work, so I don't have my hopes up and understand.

**Comment by GitHubRulesOK** [comment]
Covered this many times in issues and discussions. SumatraPDF reads characters in written sequence, not as one may expect (position order).

Adobe did not design PDF for searching simply burn single characters onto a laser drum in any order necessary.
 So letters could logically be written all down the left edge then 1 letter to the right on each line etc.
 HOWEVER other printers of the day were expected to be line printers so a PDF is usually charted in commonly horizontal bits of lines.
 They dont need to be in order or have word spaces.
 Often  a line of OCR scanning can be written backwards so English looks like it would be read by an east to west reader.

I have show an example where Abracadabra is written and searchable alphabetically  A a a a a b b c d r r (so it is wasteful on resources) but proves word based search needs be `Aaaaabbcdrr` so as to get all the letters for cut and paste.

Text accessibility extractors like Acrobat and Edge "Read Aloud" have to change letter order, into screen visual order, for Audio to make better sense. But you may sometimes hear audio start at low in a page then jump higher up the page, or even dance about in a table.

Have a read of the decoded test file in MS Notepad as also discussed here https://github.com/sumatrapdfreader/sumatrapdf/discussions/4189#discussioncomment-9136964

[Abracadabra2.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/15101657/Abracadabra2.pdf)

**Comment by renestraub** [comment]
SumatraPDF is awesome and I use it daily. The only feature I am missing right now is a search option for "whole words" only. I understand the implications of what is described in the comment of @GitHubRulesOK , but maybe one finds a way to implement this feature.

**Reply by Shohreh** [reply]
I also often need to search for whole words only.

**Reply by GitHubRulesOK** [reply]
@Shohreh 
I have shown many times PDF does not "use words" the way you may expect from a word processor.

So the sample clearly shows that plain text is NOT plain inside a PDF rendering. The reader application has to use very complex abilities to reorder letters to the same as seen on screen thus in reality there are no conventional "whole" words. 

Thus the sample shows that the word abracadabra in that file needs a very dedicated search for a letter grouping. 

So this will not work in any OTHER file as it is plaintext deliberately set to that order. But does find that whole word!
<img width="691" height="208" alt="image" src="https://github.com/user-attachments/assets/201f3a88-8675-4744-af79-d82192661a6c" />

The simplest way to try and seek a string of "letters" is add a space after the word string you are seeking and HOPE it works so here it correctly ignores the dot and finds that string
<img width="666" height="180" alt="image" src="https://github.com/user-attachments/assets/1858b485-be69-4222-a681-cb2aff17e623" />

Also if I add a  space after a letter and there is another letter such as S then it is **not a whole word**
<img width="670" height="194" alt="image" src="https://github.com/user-attachments/assets/3204b47f-a350-4871-a7d3-f477dd863198" />

**Reply by kjk** [reply]
@Shohreh whole-word search is already in the latest pre-release: https://www.sumatrapdfreader.org/prerelease

There is a "Match whole word" toggle next to "Match Case" in the find bar (and in the floating find window).

**Comment by kjk** [comment]
whole-word search is implemented in latest pre-release: https://www.sumatrapdfreader.org/prerelease

There is a "Match whole word" toggle next to "Match Case" in the find bar (and in the floating find window).
